aptos-move/mvhashmap/src/types.rs

```diff
@@ -61,3 +80,136 @@ pub enum MVModulesOutput<M, X> {
     /// if the latest entry corresponded to the module deletion.
     Module((Arc<M>, HashValue)),
 }
+
+// In order to store base vales at the lowest index, i.e. at index 0, without conflicting
+// with actual transaction index 0, the following struct wraps the index and internally
+// increments it by 1.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Debug)]
+pub(crate) struct ShiftedTxnIndex {
+    idx: TxnIndex,
+}
+
+impl ShiftedTxnIndex {
+    pub fn new(real_idx: TxnIndex) -> Self {
+        Self { idx: real_idx + 1 }
+    }
+
+    pub(crate) fn idx(&self) -> Result<TxnIndex, StorageVersion> {
```

Comment:
Definitely it needs a custom type, I liked Result because the error is indicative that it is actually a version, just "storage", as opposed to "I was supposed to sometimes have an index / version, seems I don't have it right now". Let's do in a later polishing pass so it doesn't become a distraction, but doing it nicely would be good for sure.

Reply:
added a todo